test/maps-search.test.ts

```diff
@@ -72,12 +72,26 @@ describe('Tomtom Places E2E Tests', () => {
             expect(res[0].country).toStrictEqual('Australia');
         });
 
-        it('handles error', async () => {
+        it('handles not a real address', async () => {
+            const fakeAddress = '!!%*^&%#$(&^';
+            expect(getPlaceAutocomplete(config.apiKey, fakeAddress)).rejects.toThrow('failed to retrieve data');
+        });
+
+        it('handles error when no address is passed', async () => {
             expect(getPlaceAutocomplete(config.apiKey, '')).rejects.toThrow('Address is required');
         });
 
+        it('handles error when no api key is passed', async () => {
+            const localAddress = '1 Nicholson Street Melbourne';
+            expect(getPlaceAutocomplete('', localAddress)).rejects.toThrow('API key is required');
+        });
+
         it('handles incorrect api key', async () => {
             expect(getPlaceAutocomplete('123', 'Charlotte Street')).rejects.toThrow();
         });
+        it('should map results to AutoCompleteAddressResponse', async () => {
+            const results = await mapTomTomResults(tomtomAddressResult);
+            expect(results).toEqual([addressResults]);
+        });
```

Comment:
An additional test to directly check the new `mapTomTomResults` function. In a production setting, I do feel as though `tomtomAddressResult` would require additional testing.